test/unit/geometry/geometry_is_valid.cpp

```diff
@@ -26,6 +26,218 @@ SECTION("line_string") {
 
 }
 
+SECTION("polygon ring tests") {
```

Comment:
Would suggest to add a test case for repeated points within the polygon exterior and interior rings. I would suggest that there be at least 3 points repeated in a row in this test as well. This might not be "invalid" but just a poorly constructed polygon.